File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    vector <int> ST ;
    vector <bool> visited (n,false);
    int v ; // 지금 탐색하고 있는 노드
    
    for (int i = 0 ; i < n ; i++){
        
        if (visited[i] == false){
            ST.push_back(i);
            answer++;
            visited[i] = true;
            
            while(ST.size() > 0){
                
                v = ST.back();
                ST.pop_back();
                for (int j = 0 ; j < n ; j++){
                    if (visited[j] == false && computers[v][j] == 1){
                        ST.push_back(j);
                        visited[j] = true;
                    }   
                }
            }
        }
    }

    
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    
    visited = [False] * n

    for i in range(n):
        if not visited[i]:
            stack = [i]
            visited[i]= True
                
            while stack:
                current = stack.pop()
                
                for d in range(n):
                    if not visited[d] and computers[current][d]:
                        visited[d] = True
                        stack.append(d)
                        
            answer += 1
            
            

    
    
    
    return answer